submariner/multiple-cable-drivers.md

```diff
@@ -0,0 +1,136 @@
+# Support fine-grained cluster connectivity and multiple cable drivers
+
+## Summary
+
+Submariner currently assumes full mesh connectivity by default and provisions VPN connections between all participating
+clusters. In addition, Submariner currently offers one of three cable drivers: VXLAN, IPSec or wireguard with limited
+configuration options. Only one of these cable drivers is enabled across all the clusters in a cluster set, that is all
+clusters that join a broker must use the same cable driver. The goal of this proposal is to outline an extended model
+that will support:
+
+- The selection of which clusters to connect and in what fashion (full mesh or connection to subset of clusters in
+  a clusterset).
+- The selection of one or more cable drivers in order to support different cable types between connected clusters.
+  For example, IPSec between public and private clouds but raw VXLAN between the clusters in the same cloud.
+
+There are use cases that involve supporting a mix of cable drivers or cable driver configurations within the same
+cluster set. For example, a cluster set could contain three clusters, two that are on-premise and the other
+in the cloud. Connections to the cloud cluster will need to use encryption while the on-premise clusters can use
+unencrypted connections, as they’re connected by a private link, to avoid the performance overhead of encryption.
+This could be achieved using different cable driver implementations or the same cable driver implementation with
+differing encryption configurations.
+
+For other use cases, it makes sense to offer finer-grained connectivity options. One such use case is a client-server
+scenario where multiple client clusters need to access a server cluster but the user does not want the client clusters
+to connect to one another. Finer-grained connectivity could also improve overall scalability where full mesh is not
+required.
+
+## Proposal
+
+This enhancement proposes a mechanism to allow users to specify their intent as to which clusters to connect and how to
+connect them. The default would still be a full mesh with a homogeneous cable driver.
+
```

Comment:
Do we want to auto-create a default cable driver policy as part of `subctl deploy-broker` or retain the parameters on the Submariner object?

Reply:
maybe... It will require extensions to the broker configuration - but it's certainly possible